src/test/resources/for-execution/structs.golo

```diff
@@ -80,3 +82,14 @@ function check_equals = ->
   and ImmutableCouple(1, 2) == ImmutableCouple(1, 2)
 
 function check_not_comparable = -> Couple(1, 2) < Point(1, 3)
+
+struct AList = { _holder }
+
+augment golotest.execution.Structs.types.AList {
+  function add = |this, value| -> this: _holder(): add(value)
+  function size = |this| -> this: _holder(): size()
+}
+
+function AList = -> golotest.execution.Structs.types.AList(ArrayList())
+
+function check_overload_factory = -> AList(): size()
```

Comment:
You don't have tests for the new behaviors you added (compilation error if existant function and varargs ordering in method resolution)